### PR TITLE
Add cross-builds to support other host OSs/architectures

### PR DESCRIPTION
**Pull request by androm3da**

### CODE REVIEW

**Comment by androm3da** [comment]
Not doing the cross-build for QEMU yet.  It would only make sense for the arm linux target at this point.

But once sysemu is upstreamed we can add that QEMU build to the other targets.

File: hexagon-linux-builtins.cmake [hunk 1]
```diff
@@ -0,0 +1,10 @@
+set(CMAKE_ASM_FLAGS "-G0 -mlong-calls -fno-pic" CACHE STRING "")
+
+set(LLVM_ENABLE_PER_TARGET_RUNTIME_DIR OFF CACHE BOOL "")
+set(LLVM_TARGET_TRIPLE hexagon-unknown-linux-musl CACHE STRING "")
+set(COMPILER_RT_DEFAULT_TARGET_TRIPLE hexagon-unknown-linux-musl CACHE STRING "")
+#set(COMPILER_RT_DEFAULT_TARGET_ONLY ON CACHE BOOL "")
+set(COMPILER_RT_BUILD_BUILTINS ON CACHE BOOL "")
+set(COMPILER_RT_BUILD_SANITIZERS OFF CACHE BOOL "")
```
Comment (quic-mathbern) [inline hexagon-linux-builtins.cmake:7]: This seems to be the only one that doesn't come from `build-toolchain.sh`. Just checking, was it really intended?

File: build-toolchain.sh [hunk 2]
```diff
@@ -8,6 +8,46 @@ STAMP=${1-$(date +"%Y_%b_%d")}
...
+build_llvm_clang_cross() {
+	triple=${1}
+	cd ${BASE}
+
+	EXTRA=""
+	if [[ "${triple}" =~ "windows" ]]; then
+		EXTRA="-C windows-gnu-target.cmake"
+	fi
+
+	CC="zig cc --target=${triple}" \
```
Comment (quic-mathbern) [inline build-toolchain.sh:20]: When I've tried `zig cc` with `--target=x86_64_unknown_linux_musl` before, I had problems with the "unknown" (something around the lines of this issue: https://github.com/ziglang/zig/issues/7360). Did this worked fine here for hexagon?

File: build-toolchain.sh [hunk 3]
```diff
@@ -213,6 +213,7 @@ set +x
 TOOLCHAIN_INSTALL_REL=${TOOLCHAIN_INSTALL}
 TOOLCHAIN_INSTALL=$(readlink -f ${TOOLCHAIN_INSTALL})
 TOOLCHAIN_BIN=${TOOLCHAIN_INSTALL}/x86_64-linux-gnu/bin
+TOOLCHAIN_LIB=${TOOLCHAIN_INSTALL}/x86_64-linux-gnu/lib
```
Comment (quic-mathbern) [inline build-toolchain.sh:253]: So that is where the llvm-config (or LLVMConfig.cmake) will be?

File: build-toolchain.sh (hunk 2 above)
Comment (quic-mathbern) [reply build-toolchain.sh:20]: Nevermind, just noticed that the triples passed to this function don't have the "unknown" part:

> aarch64-windows-gnu x86_64-windows-gnu aarch64-linux-gnu aarch64-macos

File: build-toolchain.sh (hunk 3 above)
Comment (androm3da) [reply build-toolchain.sh:253]: Yes - this is where `LLVMConfig.cmake` is (and not `llvm-config`).

File: hexagon-linux-builtins.cmake (hunk 1 above)
Comment (androm3da) [reply hexagon-linux-builtins.cmake:7]: > Let me check that.

Yes, `install` works instead of `install-builtins`.  Fixed.

**Pull request merged**